src/config_brain_nextroom.js

```diff
@@ -1,7 +1,10 @@
 'use strict';
 
 brain.handleNextroom = function() {
-  if (Memory.myRooms && Memory.myRooms.length < Game.gcl.level && Memory.myRooms.length < config.nextRoom.maxRooms) {
+  if (Memory.myRooms &&
+    Memory.myRooms.length < Game.gcl.level &&
+    Memory.myRooms.length < config.nextRoom.maxRooms &&
+    Memory.myRooms.length * config.nextRoom.cpuPerRoom < Game.cpu.limit) {
```

Comment:
Should be
`(Memory.myRooms.length + 1) * config...`
Otherwise it will go over the limit when colonizing the new room.

Otherwise having a single "unallocated" CPU left will allow colonization of a new room, and starve all rooms as a result.

Reply:
Not sure if I can follow.

As soon as a controller is claimed the room will end up in `myRooms`. There are some reasons how it can happen that it will claim more rooms, if you GCL is higher, but I don't see the point in that change.

Reply:
The idea of this change is to reserve at least 13 (by default) cpu per room so that each room has enough CPU to run properly, right?

What I'm noticing is that the code doesn't take the new room into account when doing this calculation (before claiming.)

If I theoretically have 14 cpu and one room, this will allow a new room to be claimed even though there would not be sufficient CPU to maintain both rooms, which I feel is unexpected behavior.

Reply:
So I have one room: 1 * 13 = 13 Cpu limit 30

Reply:
you are right